htdocs/societe/class/societe.class.php

```diff
@@ -3793,11 +3795,13 @@ public function create_from_member(Adherent $member, $socname = '', $socalias =
 		$this->client = 1; // A member is a customer by default
 		$this->code_client = ($customercode ? $customercode : -1);
 		$this->code_fournisseur = -1;
+		$this->typent_id = ($member->morphy == 'phy' ? 8 : 0); // The type of thirdparty is private (individual) if the member is human (phy)
```

Comment:
You must not use "8" here but get the id from the code 'TE_PRIVATE'. You can use method getIdFromCode... for that.

Reply:
I didn't find the method you are talking about.
Should I use the dol_function like this (from societe/card.php)
`$object->typent_code = dol_getIdFromCode($db, $object->typent_id, 'c_typent', 'id', 'code'); `
or maybe the societe method : setThirdpartyType
`$this->setThirdpartyType(8);`

Reply:
Yes, it's dol_getIdFromCode